pkg/controller/autoscaling/autoscalinggroup/setup.go

```diff
@@ -216,7 +216,10 @@ func lateInitialize(in *svcapitypes.AutoScalingGroupParameters, asg *svcsdk.Desc
 	in.CapacityRebalance = awsclients.LateInitializeBoolPtr(in.CapacityRebalance, obs.CapacityRebalance)
 	in.DefaultCooldown = awsclients.LateInitializeInt64Ptr(in.DefaultCooldown, obs.DefaultCooldown)
 	in.DefaultInstanceWarmup = awsclients.LateInitializeInt64Ptr(in.DefaultInstanceWarmup, obs.DefaultInstanceWarmup)
-	in.DesiredCapacity = awsclients.LateInitializeInt64Ptr(in.DesiredCapacity, obs.DesiredCapacity)
+	// if desiredCapacity is not set, don't update the value
+	if in.DesiredCapacity != nil {
+		in.DesiredCapacity = awsclients.LateInitializeInt64Ptr(in.DesiredCapacity, obs.DesiredCapacity)
```

Comment:
While I understand the issue this change wants to tackle I don't think it makes sense the way it is implemented: `LateInitialize` is supposed to set update the spec with the observed value *if* the former is nil.

Also, `LateInitializeInt64Ptr` only updates the value if `in.DesiredCapacity == nil`. So this change would effectively cause `in.DesiredCapacity` to be never late initialized.

I think it makes more sense to explicitly specify the desired behaviour of the controller rather then doing it implicitly through a special prop value. It could by defining a new property (i.e. `scalingMode: Managed | External`) that tells the controller to not set the value in the request to AWS during `Update`.